integrations/huggingface-transformers/README.md

```diff
@@ -62,129 +96,207 @@ output:
+
+For additional flexibility, SparseML also offers a `Trainer` class that inherits from the familiar [Transformers's Trainer](https://huggingface.co/docs/transformers/main_classes/trainer). 
+
+SparseML's `Trainer` inherits all of the functionality from the Transformers repository, but also accepts a `recipe`, which allows you to specify sparsity-related algorithms and hyperparameters. The `Trainer` parses the recipe and adjusts the training loop to apply sparsification algorithms or sparse transfer learning. As such, you can leverage Hugging Face's friendly utilities such as the Model Hub, `AutoTokenizers`, `AutoModels`, and `datasets` in concert with SparseML's sparsity-related algorithms!
+
+We create the `Trainer` and kick of a run like this:
+
+```python
+from sparseml.transformers.sparsification import Trainer, TrainingArguments
+
+def run_training(model, model_path, recipe_path, teacher, training_args, dataset, tokenizer, compute_metrics):
+    # setup training loop based on recipe passed
+    trainer = Trainer(
+        recipe=recipe_path,
+        model=model,
+        model_state_path=model_path,
+        distill_teacher=teacher,
+        metadata_args=["per_device_train_batch_size","per_device_eval_batch_size","fp16"],
+        args=training_args,
+        train_dataset=dataset["train"],
+        eval_dataset=dataset["validation"],
+        tokenizer=tokenizer,
+        compute_metrics=compute_metrics
+    )
+
+    # run training
+    trainer.train(resume_from_checkpoint=False)
+```
+
+Check out the tutorials for actual working examples using the `Trainer` class.
+
+## Quick Start: Sparse Transfer Learning
+
+### Overview
+
+Sparse Transfer is quite similiar to the typical transfer learing process used to train NLP models, where we fine-tune a pretrained checkpoint onto a smaller downstream dataset. With Sparse Transfer Learning, we simply start the fine-tuning process from a pre-sparsified checkpoint and maintain sparsity while the training process occurs.
+
+In this example, we will fine-tune a [90% pruned version of BERT](https://sparsezoo.neuralmagic.com/models/nlp%2Fmasked_language_modeling%2Fobert-base%2Fpytorch%2Fhuggingface%2Fwikipedia_bookcorpus%2Fpruned90-none) onto SST2.
 
 ### Dense Teacher Creation
 
-To enable distillation, you will first create a dense teacher model that the sparse model will learn from while transferring. **If you already have a Transformers-compatible model, you can use this as the dense teacher in place of training one from scratch.** The following command will use the dense BERT base model from the SparseZoo and fine-tune it on the SQuAD dataset, resulting in a model that achieves 88.5% F1 on the validation set: 
+To support the transfer learning process, we can (optionally) apply model distillation. To enable distillation, we first create a dense teacher model. If you already have a Transformers-compatible model, you can use this as the dense teacher in place of training one from scratch.
```

Comment:
We can (optionally) apply model distillation to support the transfer learning process